### PR TITLE
[INFRA] Add basic documentation on how to use metadata table macros

### LINKED ISSUE #838: Improve documentation to explain how to "customize" the macro calls to create metadata tables
Tempted to add some info in the contributing.md on how to formulate the call to the macros to use the metadata description contained in the schema. 
Just so that people are aware that those base description can be extended.

_Originally posted by @Remi-Gau in https://github.com/bids-standard/bids-specification/issues/804#issuecomment-889000731_

### PR DESCRIPTION
**Pull request by Remi-Gau**
fixes #838 

This is not perfect but provides some guidance on how to use the metadata table macros.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -220,6 +221,49 @@ That would look like this:
 |--------------|----------------------------------------------------------|
 | Manufacturer | Manufacturer of the equipment, for example (`"Siemens"`) | 
 
+## Using macros
+
+We use mkdocs macros to more easily make consistent the style of certain elements of the spec.
```
Comment (sappelhoff) [inline CONTRIBUTING.md:226]:
```suggestion
We use [mkdocs-macros](https://mkdocs-macros-plugin.readthedocs.io/en/latest/)
to render parts of the BIDS specification from the BIDS schema.
Macros make it easy to achieve a consistent style throughout the specification,
and changing a given macro will automatically change all appropriate paragraphs in the specification.

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -220,6 +221,49 @@ That would look like this:
...
+
+For example, this is especially convenient to generate the metadata tables by reading the content
+to put in the tables from the [yaml files](src/schema/metadata) in the [schema](src/schema/README.md).
```
Comment (sappelhoff) [inline CONTRIBUTING.md:229]:
```suggestion
For example, all tables on BIDS metadata are generated via macros that make use of data in the
[yaml files](src/schema/metadata) in the [schema](src/schema/README.md).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -220,6 +221,49 @@ That would look like this:
...
+
+The macros are written in python (see the folders [tools/schemacode](tools/schemacode) 
+and [tools/mkdocs_macros_bidsschema](tools/mkdocs_macros_bidsschema)) and are called directly in the markdown document 
+where you want output of the macro to be inserted.
```
Comment (sappelhoff) [inline CONTRIBUTING.md:233]:
```suggestion
The macros are written in Python
(see the folders [tools/schemacode](tools/schemacode) and [tools/mkdocs_macros_bidsschema](tools/mkdocs_macros_bidsschema)),
and are called directly in the Markdown document where you want the output of the macro to be inserted.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -220,6 +221,49 @@ That would look like this:
...
+   {
+      "SamplingFrequency": "REQUIRED",
+      "StartTime": "RECOMMENDED, but REQUIRED for sparse sequences",
+   }
+) }}
+```
+
+This macro will create a metadata table for the "SamplingFrequency", "StartTime",
+filling the columns of that table with the content specified in their respective yaml files
+([here](src/schema/metadata/SamplingFrequency.yaml) and [there](src/schema/metadata/StartTime.yaml)).
```
Comment (sappelhoff) [inline CONTRIBUTING.md:248]:
```suggestion
This macro will create a table for the "SamplingFrequency" and "StartTime" metadata,
filling the table with the content specified in their respective yaml files
(see [SamplingFrequency.yaml](src/schema/metadata/SamplingFrequency.yaml) and [StartTime.yaml](src/schema/metadata/StartTime.yaml)).
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -220,6 +221,49 @@ That would look like this:
...
+
+Some of the content created by the macro can specified in the macro call.
```
Comment (sappelhoff) [inline CONTRIBUTING.md:250]:
```suggestion
Some of the content created by the macro can specified in the macro call itself, as opposed to in the yaml files. 
```

**Review by sappelhoff — COMMENTED** [review]
some wording suggestions, overall LGTM

**Review by sappelhoff — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -220,6 +221,53 @@ That would look like this:
...
+      "StartTime": "RECOMMENDED, but REQUIRED for sparse sequences",
+   }
+) }}
+```
+
+This macro will create a table for the "SamplingFrequency" and "StartTime" metadata,
+filling the table with the content specified in their respective yaml files
+(see [SamplingFrequency.yaml](src/schema/metadata/SamplingFrequency.yaml) and [StartTime.yaml](src/schema/metadata/StartTime.yaml)).
+
+Some of the content created by the macro can specified in the macro call itself, as opposed to in the yaml files.
```
Comment (tsalo) [inline CONTRIBUTING.md:254]:
```suggestion
Some of the content created by the macro can be specified in the macro call itself, as opposed to in the yaml files. 
```

**Review by tsalo — CHANGES_REQUESTED** [review]
Just one minor typo. Other than that it looks great!

**Review by tsalo — APPROVED** [review]
LGTM. Thanks!

**Comment by sappelhoff** [comment]
Thanks @Remi-Gau

**Pull request merged**